lib/logstash/inputs/kafka.rb

```diff
@@ -98,7 +104,7 @@ def register
       options[:reset_beginning] = 'from-beginning'
     end # if :reset_beginning
     @kafka_client_queue = SizedQueue.new(@queue_size)
-    @consumer_group = Kafka::Group.new(options)
+    @consumer_group = @kafka_group_class.new(options)
```

Comment:
Might be something to try, but in general is good point to question the design if we are force to use the expect_any_instance_of ...  and use a dependency injection. However in this case I do understand the Kafka::Group is something that is not going to change over the time, so might be good to design the test mocking the new (if works good)